pytorch_lightning/lite/wrappers.py

```diff
@@ -157,29 +157,27 @@ def _replace_dataloader_init_method() -> Generator:
 
 
 class _LiteDataLoader:
-    def __init__(self, dataloader: Union[Iterable, DataLoader], device: Optional[torch.device] = None) -> None:
-        """The LiteDataLoader is an extension of an Iterator. It would move the data to the device automatically if
-        the device is specified.
+    def __init__(self, dataloader: DataLoader, device: Optional[torch.device] = None) -> None:
+        """The LiteDataLoader is a wrapper for the :class:`~torch.utils.data.DataLoader`. It moves the data to the
+        device automatically if the device is specified.
 
         Args:
-            dataloader: The current dataloader to be used.
+            dataloader: The dataloader to wrap
             device: The device to which the data should be moved. By default the device is `None` and no data
                 transfers will be made (identical behavior as :class:`~torch.utils.data.DataLoader`).
         """
         super().__init__()
```

Comment:
Should we maybe use `typing.cast(type(dataloader), self)` here so that user type checks still work? This could be done for models as well...

Reply:
I'm a typing noob so I can't judge if that's important or not. Will it have any implications on our typing?

Reply:
It basically sets `LiteDataLoader=Dataloader` from a typing perspective

Reply:
I couldn't figure out how this is supposed to work. For mypy to recognize the cast, I need to assign it back to the object, but here 

`self = cast(...)` won't work as we can't change the self reference.

Reply:
done in setup_dataloader directly now